src/cpp/shared/adventure/adv.cpp

```diff
@@ -313,4 +313,55 @@ void advManager::QuickInfo(int x, int y) {
   gpWindowManager->AddWindow(&tooltip, 1, -1);
   QuickViewWait();
   gpWindowManager->RemoveWindow(&tooltip);
-}
\ No newline at end of file
+}
+
+void advManager::PlayerMonsterInteract(mapCell *cell, mapCell *other, hero *player, int *window, int a1, int a2, int a3, int a4, int a5) {
+	int x;
+	int y;
+	if (cell->objType != (LOCATION_ARMY_CAMP | TILE_HAS_EVENT)) {
+		this->PlayerMonsterInteract_orig(cell, other, player, window, a1, a2, a3, a4, a5);
+		return;
+	}
+	if (GetMapCellXY(cell, &x, &y)) {
+		ScriptCallback("OnMonsterInteract", x, y);
+	}
+	if ((x < 0) && (y < 0) && (x >= gpGame->map.width) || (y >= gpGame->map.height)) {
+		this->PlayerMonsterInteract_orig(cell, other, player, window, a1, a2, a3, a4, a5);
+		return;
+	}
+	int cellIdx = y * gpGame->map.height + x;
+	this->PlayerMonsterInteract_orig(&gpGame->map.tiles[cellIdx], other, player, window, a1, a2, a3, a4, a5);
```

Comment:
Isn't the first argument just "cell"?

Reply:
Yes it is. I am simplifying this now. The overly verbose form was just for sanity's sake. Thanks for pointing it out.